### DISCUSSION TITLE
Welcome to do-mpc Discussions!

### DISCUSSION THREAD

**Discussion by do-mpc** [do-mpc/do-mpc#121] [General]
## 👋 Welcome do-mpc user!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

We prefer users to post in discussion as opposed to Issues when they have questions that are not related to problems with the code / documentation. Thank you and enjoy do-mpc!

**Comment by JonasPflaume** [comment]
Is there a reliable way to get/buy a real double inverted pendulum mechanism in europe? I visited Prof. Lucia MPC course in TU Berlin, now I'm very interested in Regelungstechnik and want to do some hands-on experiments.

**Reply by 4flixt** [reply]
Hi Jonas! Not that I know of. Maybe there are some suppliers for teaching materials but we haven't checked. I guess you would need to built it yourself. Let us know how it goes 😀

**Reply by JonasPflaume** [reply]
Thx, I haven't much background about Electronic Engineering(Circuit/Microcontroller), that's why I haven't had one jet, otherwise I should have already built one. By the way, hope you have a nice expedition in Dortmund (For us nowhere to find a suitable Master thesis without our sweat IOT team). :)

**Reply by 4flixt** [reply]
Thanks We appreciate the compliment ☺️

**Comment by isaacn-lab** [comment]
Hi all, first of all thank you for do-mpc :)
I have been trying to develop a demo EMPC using this toolbox and couldn't figure out a few things:
1. Is there a way to supply an auxiliary variable (some profile varying over prediction horizon) to the cost function (lterm)? For instance price for applying manipulated variable at certain time. In general terms this can be though as time-varying weight (similar to R matrix in r-term) that needs to be updated at each controller step. I have tried to implement it using  parameters, but it didn't really work out.
2. Is there a way to set constrains that would vary over prediction horizon?
Thank you in advance!

-Nikita
